python/rpdk/java/templates/init/guided_aws/TagHelper.java

```diff
@@ -70,27 +70,6 @@ public static Set<Tag> convertToSet(final Map<String, String> tagMap) {
             .collect(Collectors.toSet());
     }
 
-    /**
-     * generateTagsForCreate
-     *
-     * Generate tags to put into resource creation request.
-     * This includes user defined tags and system tags as well.
-     */
-    public final Map<String, String> generateTagsForCreate(final ResourceModel resourceModel, final ResourceHandlerRequest<ResourceModel> handlerRequest) {
-        final Map<String, String> tagMap = new HashMap<>();
-
-        // merge system tags with desired resource tags if your service supports CloudFormation system tags
-        tagMap.putAll(handlerRequest.getSystemTags());
-
-        if (handlerRequest.getDesiredResourceTags() != null) {
-            tagMap.putAll(handlerRequest.getDesiredResourceTags());
-        }
-
-        // TODO: get tags from resource model based on your tag property name
-        // TODO: tagMap.putAll(convertToMap(resourceModel.getTags()));
-        return Collections.unmodifiableMap(tagMap);
-    }
-
```

Comment:
Since we need to handle system tags on update as well, if system tags are enabled, `getNewDesiredTags` has to join desired stack and resource tags with system tags. Thus the code for `getNewDesiredTags` and `generateTagsForCreate` becomes identical and `getNewDesiredTags` can be used in both create and update handlers. There is no more need in a separate method to consolidate tags on create.